packages/lesswrong/components/posts/PostsItemIntroSequence.tsx

```diff
@@ -0,0 +1,237 @@
+    [theme.breakpoints.down('xs')]: {
+      width: "100%",
+    },
+    display: "flex",
+    alignItems: "center",
+  },
+  background: {
+    width: "100%",
+    background: theme.palette.panelBackground.default,
+  },
+  translucentBackground: {
+    width: "100%",
+    background: theme.palette.panelBackground.translucent,
+    backdropFilter: "blur(1px)",
+  },
+  postsItem: {
+    display: "flex",
+    position: "relative",
+    paddingTop: 10,
+    paddingBottom: 10,
+    alignItems: "center",
+    flexWrap: "nowrap",
+    flexGrow: 1,
+    [theme.breakpoints.down('xs')]: {
+      flexWrap: "wrap",
+      paddingTop: theme.spacing.unit,
+      paddingBottom: theme.spacing.unit,
+    },
+  },
+  postsItemWithImage: {
+    flexDirection: "column",
+    alignItems: "flex-start",
+    paddingTop: 15,
+    paddingBottom: 15,
+  },
+  bottomBorder: {
+    borderBottom: theme.palette.border.itemSeparatorBottom,
+  },
+  karma: {
+    width: KARMA_WIDTH,
+    justifyContent: "center",
+  },
+  title: {
+    minHeight: 26,
+    flexGrow: 1,
+    flexShrink: 1,
+    overflow: "hidden",
+    textOverflow: "ellipsis",
+    marginRight: 12,
+    [theme.breakpoints.up('sm')]: {
+      position: "relative",
+      top: 3,
+    },
+    [theme.breakpoints.down('xs')]: {
+      order:-1,
+      height: "unset",
+      maxWidth: "unset",
+      width: "100%",
+      paddingRight: theme.spacing.unit,
+    },
+    '&:hover': {
+      opacity: 1,
+    },
+  },
+  author: {
+    overflow: "hidden",
+    whiteSpace: "nowrap",
+    textOverflow: "ellipsis",
+    marginRight: theme.spacing.unit*1.5,
+    zIndex: theme.zIndexes.postItemAuthor,
+    [theme.breakpoints.down('xs')]: {
+      justifyContent: "flex-end",
+      width: "unset",
+      marginLeft: 0,
+      flex: "unset",
+    },
+  },
+  mobileSecondRowSpacer: {
+    [theme.breakpoints.up('sm')]: {
+      display: "none",
+    },
+    flexGrow: 1,
+  },
+  sequenceImage: {
+    position: "absolute",
+    height: "100%",
+    marginTop: 0,
+    marginBottom: 0,
+    overflow: 'hidden',
+    right: 0,
+    bottom: 0,
+
+    // Overlay a white-to-transparent gradient over the image
+    "&:after": {
+      content: "''",
+      position: "absolute",
+      width: "100%",
+      height: "100%",
+      left: 0,
+      top: 0,
+      background: `linear-gradient(to right, ${theme.palette.panelBackground.default} 0%, ${theme.palette.panelBackground.translucent2} 30%, transparent 100%)`,
+    },
+  },
+  sequenceImageImg: {
+    height: "100%",
+    width: 'auto',
+  },
+  dense: {
+    paddingTop: 7,
+    paddingBottom:8,
+  },
+})
+
+const cloudinaryCloudName = cloudinaryCloudNameSetting.get()
+
+const PostsItemIntroSequence = ({
+  // post: The post displayed.
+  post,
+  // sequenceId, chapter: If set, these will be used for making a nicer URL.
+  sequenceId,
+  chapter,
+  sequence,
+  showBottomBorder=true,
+  // dense: (bool) Slightly reduce margins to make this denser. Used on the
+  // All Posts page.
```

Comment:
This might be copy-pasta that can be removed. If it's not, apply it in proper docstring form (`/** ... */`) to the type definition.

Reply:
Same for the others.